### PR TITLE
cabelo@opensuse.org - Build in openSUSE:compatible with gcc7

### PR DESCRIPTION
**Pull request by cabelo**

### CODE REVIEW

File: ggml-quants.c [hunk 1]
```diff
@@ -52,6 +52,11 @@
 #define UNUSED GGML_UNUSED
 
 #define MM256_SET_M128I(a, b) _mm256_insertf128_si256(_mm256_castsi128_si256(b), (a), 1)
+#ifdef __GNUC__ 
+#if __GNUC__ < 8
+#define _mm256_set_m128i(a, b) _mm256_permute2f128_si256(_mm256_castsi128_si256(a), _mm256_castsi128_si256(b), 2)
+#endif
+#endif
```
Comment (cebtenzzre) [inline ggml-quants.c:59]: clang only claims to be GCC 4.2 according to \_\_GNUC\_\_ and \_\_GNUC\_MINOR\_\_, so this should be:
```suggestion
#if defined(__GNUC__) && __GNUC__ < 8 && !defined(__clang__)
#define _mm256_set_m128i(a, b) _mm256_permute2f128_si256(_mm256_castsi128_si256(a), _mm256_castsi128_si256(b), 2)
#endif
```

Comment (cabelo) [reply ggml-quants.c]: You are absolutely right, thank you for your patience. Done!

Comment (cebtenzzre) [reply ggml-quants.c]: > But these calls to _mm256_set_m128i were only added in #&#8203;5187, previously only MM256_SET_M128I was used - you should try removing this macro and using that one instead.

cc @ikawrakow

**Review by ikawrakow — APPROVED** [review]
Does it really work with any GCC version earlier than 7?

**Comment by cabelo** [comment]
> Does it really work with any GCC version earlier than 7?

@ikawrakow I can't answer this, do you want me to set the condition to only meet version 7?

**Comment by ikawrakow** [comment]
> do you want me to set the condition to only meet version 7?

I have no strong feelings about this, was just surprised that a) someone is still using GCC 7, and b) the condition was less `< 8`, so included even earlier versions.

What is the use case for GCC 7 in the year 2024 anyway?

**Comment by cebtenzzre** [comment]
I will mention again that it seems like this PR is adding yet another implementation of _mm256_set_m128i 
 (MM256_SET_M128I is defined above). I don't know which is preferred or why one is conditionally emulated and the other is unconditionally emulated.

**Comment by cabelo** [comment]
> > do you want me to set the condition to only meet version 7?
> 
> I have no strong feelings about this, was just surprised that a) someone is still using GCC 7, and b) the condition was less `< 8`, so included even earlier versions.
> 
> What is the use case for GCC 7 in the year 2024 anyway?
Hi super @ikawrakow 
Because I intended to include llama.cpp natively in openSUSE and SUSE Enterprise, and the distribution natively comes with version 7

**Comment by cabelo** [comment]
@ikawrakow Do I need to do anything else?

**Comment by slaren** [comment]
I think we should use the `MM256_SET_M128I` macro everywhere instead of adding another workaround, it was added precisely for this purpose. https://github.com/ggerganov/llama.cpp/pull/2405

**Pull request closed without merging**